ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/NamespaceView/NamespaceViewPage.tsx

```diff
@@ -138,8 +142,27 @@ function NamespaceViewPage() {
         searchFilter,
     });
 
+    function onSearch({ category, value, action }: OnSearchPayload) {
+        const selectedSearchFilter = searchValueAsArray(searchFilter[category]);
+
+        const newFilter = {
+            ...searchFilter,
+            [category]:
+                action === 'ADD'
+                    ? uniq([...selectedSearchFilter, value])
+                    : selectedSearchFilter.filter((oldValue) => value !== oldValue),
+        };
+
+        if (action === 'ADD') {
+            // TODO - Add analytics tracking
+        }
+
+        setSearchFilter(newFilter);
+        onFilterChange();
+    }
```

Comment:
I recently made a function called `onURLSearch` in the `CompoundSearchFilter/utils` file. I think we can take what you did here and use it there. So we could ideally update this function to work like this:

```
const onSearch = ({ category, value, action }: OnSearchPayload) => {
    if (action === 'ADD') {
        // TODO - Add analytics tracking
    }
    onURLSearch(searchFilter, setSearchFilter, payload);
    onFilterChange();
};
```

Reply:
Ah nice - I have an implementation like this in the AdvancedFiltersToolbar as well. I can refactor these both out in a follow up.